### ISSUE TITLE
Unqualified references in generated enclosed types fail to compile

### ISSUE THREAD

**Issue by jfancher** [immutables/immutables#408]
````java
@Value.Enclosing
public interface Foo {
    String value();

    @Value.Immutable
    interface Bar extends Foo {}
}
````

This fails to compile because several generated methods refer to an unqualified `Bar` which is taken to refer to `Foo.Bar` instead of `ImmutableFoo.Bar`. I don't understand why this is the case... unless I am misunderstanding something about Java shadowing rules it seems like a compiler bug (although it happens both in Eclipse and javac). Whatever the case, it seems straightforward enough to fix/work around, by qualifying references to the type, which is already done elsewhere such as in the Builder.

**Comment by elucash** [comment]
Thank you for revealing this! hope it would be easy to make references properly qualified.

**Comment by elucash** [comment]
released in 2.2.11 Thank you!
